go/kbhttp/srv.go

```diff
@@ -152,11 +155,12 @@ func (h *Srv) Addr() (string, error) {
 }
 
 // Stop stops listening on the server's listener source.
-func (h *Srv) Stop() {
+func (h *Srv) Stop() <-chan struct{} {
 	h.Lock()
 	defer h.Unlock()
 	if h.server != nil {
 		h.server.Close()
 		h.server = nil
 	}
+	return h.doneCh
```

Comment:
this will hang if the server never started

Reply:
That should be fine. Right now we only block in tests.

Reply:
could close before returning if server is nil